pyanaconda/kickstart.py

```diff
@@ -1270,12 +1271,29 @@ def execute(self, storage, ksdata, instClass):
             self.passphrase = self.passphrase or storage.encryption_passphrase
 
             cert = getEscrowCertificate(storage.escrow_certificates, self.escrowcert)
+
```

Comment:
You have this identical code 3 times in this file, please make that a method instead.

Reply:
The shared code is in the `get_pbkdf_args` method and I can put there also these lines:

```
if pbkdf_args and not luks_data.pbkdf_args:
    luks_data.pbkdf_args = pbkdf_args
```

Otherwise, I don't like it as well, but these execute methods are awful already and need to be refactorized. One weird method just for LUKS will not help with it. If you look above, you can see that the code for a passphrase and a certificate is also identical.

Reply:
The question is if you don't want to start with the refactorization on this point :). But I agree this is not something for NACK in this case.